### PR TITLE
feat(environments): Add environment option to alert rules

### PR DESCRIPTION
**Pull request by lynnagara**

### CODE REVIEW

File: src/sentry/static/sentry/app/views/ruleEditor/index.jsx [hunk 1]
```diff
@@ -149,20 +151,40 @@ const RuleEditor = createReactClass({
     return !!error[field];
   },
 
+  handleEnvironmentChange(val) {
+    // If 'All Environments' is selected remove the environment property from the data
+    if (val === 'all') {
+      this.setState(state => {
+        const rule = {...state.rule};
+        delete rule.environment;
+        return {rule};
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/ruleEditor/index.jsx]: what about:
```
let {environment, ...rule} = state.rule;
return {rule};
```

File: src/sentry/static/sentry/app/views/ruleEditor/index.jsx [hunk 2]
```diff
@@ -149,20 +151,40 @@ const RuleEditor = createReactClass({
     return !!error[field];
   },
 
+  handleEnvironmentChange(val) {
+    // If 'All Environments' is selected remove the environment property from the data
+    if (val === 'all') {
+      this.setState(state => {
+        const rule = {...state.rule};
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/ruleEditor/index.jsx]: Can we keep using `let` until we bulk reformat codebase and add a lint rule?

**Review by billyvg — APPROVED** [review]

File: src/sentry/static/sentry/app/views/ruleEditor/index.jsx (hunk 1 above)
Comment (lynnagara) [reply src/sentry/static/sentry/app/views/ruleEditor/index.jsx]: I don't think that's better. It breaks our linting rules.

File: src/sentry/static/sentry/app/views/ruleEditor/index.jsx (hunk 2 above)
Comment (lynnagara) [reply src/sentry/static/sentry/app/views/ruleEditor/index.jsx]: I've been using const pretty consistently for new stuff and noticed others have too. I don't think there's a good reason to insist on let now just so we can add a lint rule soon and reformat this line later.

**Pull request merged**